packages/form-js-viewer/src/render/components/form-fields/parts/ChildrenRenderer.js

```diff
@@ -0,0 +1,126 @@
+import { useContext, useState } from 'preact/hooks';
+
+import useService from '../../../hooks/useService';
+
+import FormField from '../../FormField';
+
+import { FormRenderContext } from '../../../context';
+
+export default function ChildrenRenderer(props) {
+
+  const {
+    Children
+  } = useContext(FormRenderContext);
+
+  const { field, Empty } = props;
+
+  const { id } = field;
+
+  const repeatRenderManager = useService('repeatRenderManager', false);
```

Comment:
I like this a lot as it opens up the possibility of replacing the complete repeating mechanism via extension. We could consider the same for the simple rendering and rows rendering, e.g. to replace Carbon Grid with something else.

Just an idea, no call to action 🙂